### PR TITLE
Feature/compose sample

### LINKED ISSUE #59: [Feature] | Add example module in compose
**Which kind of problem this feature would address? Please describe.**
Provide an example of how to use the library with compose.

**Describe the solution you'd like**
A screen with fields where you can see each type of error handled in compose.

### PR DESCRIPTION
**Pull request by ximenaperez**
#### ISSUE[#59]
* closes #59

---

#### Description
* Add code sample for android compose

---

#### Tasks
* Add compose screen for login

---

#### Preview
Screenshot_20230915_155058

### CODE REVIEW

File: exampleappandroidcompose/build.gradle.kts
Comment (hrodrick) [inline exampleappandroidcompose/build.gradle.kts:1]: Please review the module's name, try to keep the same naming style as the other modules: ExampleApp Android Compose

image

File: exampleappandroidcompose/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (hrodrick) [inline exampleappandroidcompose/.gitignore:1]: Please add an empty line at the end of this file

File: exampleappandroidcompose/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    implementation("androidx.lifecycle:lifecycle-runtime-ktx:2.6.1")
+    implementation("androidx.activity:activity-compose:1.7.2")
+    implementation(platform("androidx.compose:compose-bom:2022.10.00"))
+    implementation("androidx.compose.ui:ui")
+    implementation("androidx.compose.ui:ui-graphics")
+    implementation("androidx.compose.ui:ui-tooling-preview")
+    implementation("androidx.compose.material3:material3")
+    debugImplementation("androidx.compose.ui:ui-tooling")
+    debugImplementation("androidx.compose.ui:ui-test-manifest")
+}
```
Comment (hrodrick) [inline exampleappandroidcompose/build.gradle.kts:65]: Please add an empty line at the end of this file

File: exampleappandroidcompose/src/androidTest/java/com/example/exampleappandroidcompose/ExampleInstrumentedTest.kt
Comment (hrodrick) [inline exampleappandroidcompose/src/androidTest/java/com/example/exampleappandroidcompose/ExampleInstrumentedTest.kt:1]: This file doesn't seem very useful, please remove

File: exampleappandroidcompose/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            android:theme="@style/Theme.FlowForms">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+    </application>
+
+</manifest>
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/AndroidManifest.xml:24]: Please add an empty line at the end of this file

File: exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package com.example.exampleappandroidcompose
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/MainActivity.kt:8]: I believe we don't need many of the imports here. Please use the `optimize imports` tool. I also recommend to enable it in the Android Studios' settings.
image

File: exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/screens/SignUpScreen.kt [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            SignUpFormUiState(),
+            onNameChanged = {},
+            onEmailChanged = {},
+            onPasswordChanged = {},
+            onPasswordConfirmationChanged = {},
+            onTermsAcceptedChange = {},
+            onSignupButtonClicked = {}
+        )
+    }
+}
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/screens/SignUpScreen.kt:130]: Please add an empty line at the end of this file

File: exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/theme/Color.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+import androidx.compose.ui.graphics.Color
+
+val Purple80 = Color(0xFFD0BCFF)
+val PurpleGrey80 = Color(0xFFCCC2DC)
+val Pink80 = Color(0xFFEFB8C8)
+
+val Purple40 = Color(0xFF6650a4)
+val PurpleGrey40 = Color(0xFF625b71)
+val Pink40 = Color(0xFF7D5260)
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/theme/Color.kt:11]: Same

File: exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/theme/Dimension.kt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.example.exampleappandroidcompose.ui.theme
+
+import androidx.compose.ui.unit.dp
+
+object Padding {
+    val small = 5.dp
+    val medium = 10.dp
+    val big = 15.dp
+    val huge = 20.dp
+}
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/theme/Dimension.kt:10]: Same

File: exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/theme/Type.kt [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    ),
+    labelSmall = TextStyle(
+        fontFamily = FontFamily.Default,
+        fontWeight = FontWeight.Medium,
+        fontSize = 11.sp,
+        lineHeight = 16.sp,
+        letterSpacing = 0.5.sp
+    )
+    */
+)
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/java/com/example/exampleappandroidcompose/ui/theme/Type.kt:34]: Same

File: exampleappandroidcompose/src/main/res/values/colors.xml [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="purple_200">#FFBB86FC</color>
+    <color name="purple_500">#FF6200EE</color>
+    <color name="purple_700">#FF3700B3</color>
+    <color name="teal_200">#FF03DAC5</color>
+    <color name="teal_700">#FF018786</color>
+    <color name="black">#FF000000</color>
+    <color name="white">#FFFFFFFF</color>
+</resources>
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/res/values/colors.xml:10]: Same

File: exampleappandroidcompose/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    <string name="required_field">This field is required</string>
+    <string name="invalid_email">This email is invalid</string>
+    <string name="email_already_exist">This email is already in use</string>
+    <string name="email_is_available">Email is available</string>
+    <string name="min_length">This field should have %1$d characters</string>
+    <string name="password_match">The passwords don\'t match</string>
+    <string name="account_registered">Account registered</string>
+    <string name="review_form">Please review the form</string>
+    <string name="incorrect_form_warning">Form is incorrect but you can try to continue</string>
+</resources>
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/res/values/strings.xml:19]: Same

File: exampleappandroidcompose/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -0,0 +1,19 @@
+<resources>
+    <string name="app_name">Example App Android Compose</string>
+    <string name="name">Name</string>
+    <string name="signup">Sign Up</string>
+    <string name="email">Email</string>
+    <string name="password">Password</string>
+    <string name="confirm_password">Confirm Password</string>
+    <string name="confirmation">Confirm data</string>
+    <string name="continue_btn">Continue</string>
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/res/values/strings.xml:9]: Looks like this String is not used, please remove

File: exampleappandroidcompose/src/main/res/values/themes.xml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="Theme.FlowForms" parent="android:Theme.Material.Light.NoActionBar" />
+</resources>
```
Comment (hrodrick) [inline exampleappandroidcompose/src/main/res/values/themes.xml:5]: Please add an empty line at the end of this file

File: exampleappandroidcompose/src/main/res/values/colors.xml (hunk 9 above)
Comment (hrodrick) [inline exampleappandroidcompose/src/main/res/values/colors.xml:10]: It looks like these colors are not used, please remove

File: exampleappandroidcompose/src/test/java/com/example/exampleappandroidcompose/ExampleUnitTest.kt
Comment (hrodrick) [inline exampleappandroidcompose/src/test/java/com/example/exampleappandroidcompose/ExampleUnitTest.kt:1]: This file doesn't seem to be needed, please remove

**Review by hrodrick — CHANGES_REQUESTED** [review]
Nice work! Just left some comments

**Review by hrodrick — APPROVED** [review]

File: ExampleApp Android Compose/build.gradle.kts [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+        resources {
+            excludes += "/META-INF/{AL2.0,LGPL2.1}"
+        }
+    }
+}
+
+dependencies {
+    implementation(project(":shared"))
+    implementation(Dependencies.appComposeLibraries)
+    implementation(platform("androidx.compose:compose-bom:2022.10.00"))
```
Comment (marcossalto) [inline ExampleApp Android Compose/build.gradle.kts:55]: I think we could move all of those hardcoded libraries to Dependencies object.

File: ExampleApp Android Compose/proguard-rules.pro [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+#   public *;
+#}
+
+# Uncomment this to preserve the line number information for
+# debugging stack traces.
+#-keepattributes SourceFile,LineNumberTable
+
+# If you keep the line number information, uncomment this to
+# hide the original source file name.
+#-renamesourcefileattribute SourceFile
```
Comment (marcossalto) [inline ExampleApp Android Compose/proguard-rules.pro:21]: Please add an empty line at EOF

**Review by marcossalto — APPROVED** [review]
Well done! I left a couple of minor comments

**Pull request merged**